Assets/Scripts/PlayerFlash.cs

```diff
@@ -0,0 +1,72 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+public class PlayerFlash : MonoBehaviour
+{
+    [SerializeField]
+    private Camera playerView;
+    [SerializeField]
+    private GameObject enemyObject;
+    [SerializeField]
+    private LayerMask groundLayer, enemyLayer;
+    [SerializeField]
+    private Light flashPointLight;
+    private Animator flashAnimator;
+
+    //View Variables
+    private bool obstructed;
+    private bool hitEnemy;
+    private Vector3 toEnemy;
+
+    //Flash Variables
+    [SerializeField]
+    private float flashRange;
+    [SerializeField]
+    private float stunDuration;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        flashAnimator = flashPointLight.GetComponent<Animator>();
+    }
+
+    private void FixedUpdate()
+    {
+        obstructed = Physics.Raycast(transform.position, toEnemy, toEnemy.magnitude, groundLayer);
+        hitEnemy = Physics.Raycast(transform.position, toEnemy, flashRange, enemyLayer);
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        toEnemy = enemyObject.transform.position - transform.position;
+        Vector3 enemyToCam = playerView.WorldToViewportPoint(enemyObject.transform.position);
+
+        bool leftClick = Input.GetMouseButtonDown(0);
+
+        if (leftClick)
```

Comment:
This works since we are allowing both input systems but it would be best to standardize and use the new input system.